<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание-проекта" data-toc-modified-id="Описание-проекта-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание проекта</a></span></li><li><span><a href="#Изучение-и-подготовка-данных" data-toc-modified-id="Изучение-и-подготовка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Изучение и подготовка данных</a></span><ul class="toc-item"><li><span><a href="#Регистр-столбцов" data-toc-modified-id="Регистр-столбцов-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Регистр столбцов</a></span></li><li><span><a href="#Пропуски" data-toc-modified-id="Пропуски-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Пропуски</a></span><ul class="toc-item"><li><span><a href="#столбец-critic_score" data-toc-modified-id="столбец-critic_score-2.2.1"><span class="toc-item-num">2.2.1&nbsp;&nbsp;</span>столбец critic_score</a></span></li><li><span><a href="#столбец-user_score" data-toc-modified-id="столбец-user_score-2.2.2"><span class="toc-item-num">2.2.2&nbsp;&nbsp;</span>столбец <em>user_score</em></a></span></li><li><span><a href="#столбец-rating" data-toc-modified-id="столбец-rating-2.2.3"><span class="toc-item-num">2.2.3&nbsp;&nbsp;</span>столбец <em>rating</em></a></span></li><li><span><a href="#столбцы-year_of_release,-name,-genre" data-toc-modified-id="столбцы-year_of_release,-name,-genre-2.2.4"><span class="toc-item-num">2.2.4&nbsp;&nbsp;</span>столбцы <em>year_of_release, name, genre</em></a></span></li><li><span><a href="#Расчёт-и-добавление-в-данные-суммарных-продаж" data-toc-modified-id="Расчёт-и-добавление-в-данные-суммарных-продаж-2.2.5"><span class="toc-item-num">2.2.5&nbsp;&nbsp;</span>Расчёт и добавление в данные суммарных продаж</a></span></li></ul></li></ul></li><li><span><a href="#Исследовательский-анализ-данных" data-toc-modified-id="Исследовательский-анализ-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Исследовательский анализ данных</a></span><ul class="toc-item"><li><span><a href="#Выход-игр-по-годам" data-toc-modified-id="Выход-игр-по-годам-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Выход игр по годам</a></span></li><li><span><a href="#Продажи-по-платформам" data-toc-modified-id="Продажи-по-платформам-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Продажи по платформам</a></span></li><li><span><a href="#Платформы-с-наибольшими-суммарными-продажами" data-toc-modified-id="Платформы-с-наибольшими-суммарными-продажами-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Платформы с наибольшими суммарными продажами</a></span></li><li><span><a href="#Динамика-продаж" data-toc-modified-id="Динамика-продаж-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Динамика продаж</a></span></li><li><span><a href="#Диаграммы-размаха-по-глобальным-продажам-игр-в-разбивке-по-платформам" data-toc-modified-id="Диаграммы-размаха-по-глобальным-продажам-игр-в-разбивке-по-платформам-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Диаграммы размаха по глобальным продажам игр в разбивке по платформам</a></span></li><li><span><a href="#Влияние-на-продажи-внутри-популярной-платформы-отзывов-пользователей-и-критиков" data-toc-modified-id="Влияние-на-продажи-внутри-популярной-платформы-отзывов-пользователей-и-критиков-3.6"><span class="toc-item-num">3.6&nbsp;&nbsp;</span>Влияние на продажи внутри популярной платформы отзывов пользователей и критиков</a></span><ul class="toc-item"><li><span><a href="#Диаграммы-рассеяния" data-toc-modified-id="Диаграммы-рассеяния-3.6.1"><span class="toc-item-num">3.6.1&nbsp;&nbsp;</span>Диаграммы рассеяния</a></span></li><li><span><a href="#Матрицы-корреляции" data-toc-modified-id="Матрицы-корреляции-3.6.2"><span class="toc-item-num">3.6.2&nbsp;&nbsp;</span>Матрицы корреляции</a></span></li></ul></li><li><span><a href="#Распределение-игр-по-жанрам" data-toc-modified-id="Распределение-игр-по-жанрам-3.7"><span class="toc-item-num">3.7&nbsp;&nbsp;</span>Распределение игр по жанрам</a></span></li></ul></li><li><span><a href="#Портрет-пользователя-каждого-региона" data-toc-modified-id="Портрет-пользователя-каждого-региона-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Портрет пользователя каждого региона</a></span><ul class="toc-item"><li><span><a href="#Самые-популярные-платформы--(топ-5)" data-toc-modified-id="Самые-популярные-платформы--(топ-5)-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Самые популярные платформы  (топ-5)</a></span></li><li><span><a href="#Самые-популярные-жанры-(топ-5)" data-toc-modified-id="Самые-популярные-жанры-(топ-5)-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Самые популярные жанры (топ-5)</a></span></li><li><span><a href="#Влияет-ли-рейтинг-ESRB-на-продажи-в-отдельном-регионе" data-toc-modified-id="Влияет-ли-рейтинг-ESRB-на-продажи-в-отдельном-регионе-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Влияет ли рейтинг ESRB на продажи в отдельном регионе</a></span></li></ul></li><li><span><a href="#Проверка-гипотез" data-toc-modified-id="Проверка-гипотез-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Проверка гипотез</a></span><ul class="toc-item"><li><span><a href="#Средние-пользовательские-рейтинги-платформ-Xbox-One-и-PC-одинаковые" data-toc-modified-id="Средние-пользовательские-рейтинги-платформ-Xbox-One-и-PC-одинаковые-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Средние пользовательские рейтинги платформ Xbox One и PC одинаковые</a></span><ul class="toc-item"><li><span><a href="#Тест-Левена" data-toc-modified-id="Тест-Левена-5.1.1"><span class="toc-item-num">5.1.1&nbsp;&nbsp;</span>Тест Левена</a></span></li><li><span><a href="#Проверка-гипотезы" data-toc-modified-id="Проверка-гипотезы-5.1.2"><span class="toc-item-num">5.1.2&nbsp;&nbsp;</span>Проверка гипотезы</a></span></li></ul></li><li><span><a href="#Средние-пользовательские-рейтинги-жанров-Action-и-Sports-разные" data-toc-modified-id="Средние-пользовательские-рейтинги-жанров-Action-и-Sports-разные-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>Средние пользовательские рейтинги жанров Action и Sports разные</a></span><ul class="toc-item"><li><span><a href="#Тест-Левена" data-toc-modified-id="Тест-Левена-5.2.1"><span class="toc-item-num">5.2.1&nbsp;&nbsp;</span>Тест Левена</a></span></li><li><span><a href="#Проверка-гипотезы" data-toc-modified-id="Проверка-гипотезы-5.2.2"><span class="toc-item-num">5.2.2&nbsp;&nbsp;</span>Проверка гипотезы</a></span></li></ul></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

СБОРНЫЙ ПРОЕКТ</font><a name="К."></a>

# Описание проекта

 <font size=2>
<b>Описание проекта</b><br/>    

Вы работаете в интернет-магазине «Стримчик», который продаёт по всему миру компьютерные игры. Из открытых источников доступны исторические данные о продажах игр, оценки пользователей и экспертов, жанры и платформы (например, Xbox или PlayStation). Вам нужно выявить определяющие успешность игры закономерности. Это позволит сделать ставку на потенциально популярный продукт и спланировать рекламные кампании.<br/>
Перед вами данные до 2016 года. Представим, что сейчас декабрь 2016 г., и вы планируете кампанию на 2017-й. <b>Нужно отработать принцип работы с данными.</b> Неважно, прогнозируете ли вы продажи на 2017 год по данным 2016-го или же 2027-й — по данным 2026 года.

<i>В наборе данных попадается аббревиатура ESRB (Entertainment Software Rating Board) — это ассоциация, определяющая возрастной рейтинг компьютерных игр. ESRB оценивает игровой контент и присваивает ему подходящую возрастну|ю категорию, например, «Для взрослых», «Для детей младшего возраста» или «Для подростков»</i>.

<font size=2>
<b>Описание данных</b><br/> 
    
| `Name` 	| название игры 	|
|---	|:---	|
| `Platform` 	| платформа 	|
| `Year_of_Release` 	| год выпуска 	|
| `Genre` 	| жанр игры 	|
| `NA_sales` 	| продажи в Северной Америке (миллионы проданных копий) 	|
| `EU_sales` 	| продажи в Европе (миллионы проданных копий) 	|
| `JP_sales` 	| продажи в Японии (миллионы проданных копий) 	|
| `Other_sales` 	| продажи в других странах (миллионы проданных копий) 	|
| `Critic_Score` 	| оценка критиков (максимум 100) 	|
| `User_Score` 	| оценка пользователей (максимум 10) 	|
| `Rating` 	| рейтинг от организации ESRB (англ. Entertainment Software Rating Board). 	|
    
    *Данные за 2016 год могут быть неполными.

# Изучение и подготовка данных
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [1]:
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
from pandas.plotting import scatter_matrix
import pandas as pd
from scipy import stats as st
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot
import plotly
from plotly.subplots import make_subplots
import plotly.graph_objs as go
import plotly.io as pio
pio.templates.default = "ggplot2"
sns.set_style(
    "darkgrid", {"grid.color": "midnightblue", "grid.linestyle": ":"})
sns.set(font_scale=1.5)
# pd.set_option('display.float_format', '{:,.2f}'.format)
pd.set_option('display.max_columns', None)
init_notebook_mode(connected=True)

games = pd.read_csv('games.csv')
#!не забудь расскоментить строчку ниже перед отправкой на проверку!
#games = pd.read_csv('/datasets/games.csv')
games.info()
(games.sample(15).style.set_properties(
    **{'text-align': 'center'}).highlight_null(null_color='lightcoral'))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
Name               16713 non-null object
Platform           16715 non-null object
Year_of_Release    16446 non-null float64
Genre              16713 non-null object
NA_sales           16715 non-null float64
EU_sales           16715 non-null float64
JP_sales           16715 non-null float64
Other_sales        16715 non-null float64
Critic_Score       8137 non-null float64
User_Score         10014 non-null object
Rating             9949 non-null object
dtypes: float64(6), object(5)
memory usage: 1.4+ MB


## Регистр столбцов
<font size="2"><i>([к началу проекта](#К.))</i></font>


<font size=2>
Первое, что бросается в глаза - <b>"НАзвания столбцов"</b>, сразу, для нашего же дальнейшего удобства, приведём их названия к нижнему регистру:

In [2]:
games.columns = games.columns.str.lower()
games.sample(3)

name platform  year_of_release    genre  \
15294                           Azada      3DS           2012.0   Puzzle   
1040   Call of Duty: Modern Warfare 3       PC           2011.0  Shooter   
14998                         ZooCube       GC           2002.0   Puzzle   

       na_sales  eu_sales  jp_sales  other_sales  critic_score user_score  \
15294      0.00      0.02       0.0         0.00           NaN        tbd   
1040       0.41      0.98       0.0         0.32          78.0        2.5   
14998      0.02      0.00       0.0         0.00          67.0        6.1   

      rating  
15294      E  
1040       M  
14998      E

## Пропуски
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [3]:
prcnt_miss_value = pd.DataFrame(games.isna().mean()).reset_index()
prcnt_miss_value.set_axis(
    ['column_name', 'missing_value'], axis='columns', inplace=True)
prcnt_miss_value.sort_values(
    by='missing_value', ascending=False).reset_index(
    drop=True).style.bar(
    subset=['missing_value'], color='teal', vmin=0, width=100).set_caption(
    '% пропусков в исходных данных')

In [4]:
prcnt_mean_value = prcnt_miss_value['missing_value'].mean()
prcnt_y_of_r = len(games.axes[0]) - games['year_of_release'].count()
print("Средний процент пропущенных значений в исходных данных: {:.2%}".format(
    prcnt_mean_value))
print("Кол-во прорущенных значений в столбце year_of_release", prcnt_y_of_r)

Средний процент пропущенных значений в исходных данных: 12.14%
Кол-во прорущенных значений в столбце year_of_release 269


<font size=2>
Больше всего пропусков в столбцах <b>Critic_Score, Rating и User_Score</b> - <b>51%, 40,5% и 40%</b> соответственно. В столбце <b>Year_of_Release</b> пропусков не так много - пропущено 269 значений <b>(1,6%)</b>. Также есть единичные пропуски в столбцах <b>Name</b> и <b>Genre</b>. <br/> Посмотрим подробнее на строки с отсутсвующими оценками:

In [5]:
games.query(
    'critic_score.isnull() & rating.isnull() & user_score.isnull()').sample(3)

name platform  year_of_release      genre  na_sales  \
8262   Power Pro Kun Pocket       GB           1999.0     Sports       0.0   
13403         Amnesia Later      PSP           2012.0       Misc       0.0   
15409           Last Escort      PS2           2006.0  Adventure       0.0   

       eu_sales  jp_sales  other_sales  critic_score user_score rating  
8262        0.0      0.17          0.0           NaN        NaN    NaN  
13403       0.0      0.05          0.0           NaN        NaN    NaN  
15409       0.0      0.02          0.0           NaN        NaN    NaN

<font size=2>
Никакой зависимости пропусков от года выхода игры, жанра, платформы или продаж установить не удалось, пропуски вариативны. Идём по порядку, и рассматриваем пропуски в  данных в порядке их уменьшения:

### столбец critic_score
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [6]:
games['critic_score'].unique()

array([76., nan, 82., 80., 89., 58., 87., 91., 61., 97., 95., 77., 88.,
       83., 94., 93., 85., 86., 98., 96., 90., 84., 73., 74., 78., 92.,
       71., 72., 68., 62., 49., 67., 81., 66., 56., 79., 70., 59., 64.,
       75., 60., 63., 69., 50., 25., 42., 44., 55., 48., 57., 29., 47.,
       65., 54., 20., 53., 37., 38., 33., 52., 30., 32., 43., 45., 51.,
       40., 46., 39., 34., 35., 41., 36., 28., 31., 27., 26., 19., 23.,
       24., 21., 17., 22., 13.])

<font size=2>
Поскольку пропусков очень много, заменять их на среднее или медианное значение, даже с какой-либо группировкой будет некорректно, в игровой индустрии взлёты и падения серий игр нередки, как и нередки случаи высоких оценок критиков при низкой оценке игроков, каждое значение уникально, поэтому, чтобы не искажать данные заполним пропуски характерной и узнаваемой заглушкой -  <i>"-1"</i>, поскольку в исходных данных не встречается игр с отрицательным рейтингом.

In [7]:
games['critic_score'] = games['critic_score'].fillna(-1)

### столбец *user_score*
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [8]:
games['user_score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

<font size=2>Помимо неопределённых значений, как и в столбце <b>critic_score</b> есть значения <i>'tbd'</i>. Проведя поиск узнаём значение аббревиатуры - <i>To Be Determined</i>, согласно [FAQ Metacritic](https://www.metacritic.com/faq#item13) такое значение  получает какое-либо произведение (фильм, игра, телешоу, альбом) не имеющие как минимум 4 оценок.<br/>
Данные у нас представлены практически по 17 тысячам игр начиная в период с 1980 по 2016 год, можем предположить, что отсутствие оценок может быть связано как и со старостью игры, её узконаправленным уникальном жанром, популярностью в каком-то отдельном небольшом регионе, фактически в нашем случае <i>'tbd'</i> и означает <i>'nan'</i>, поэтому произведём замену затем также как и в <b>critic_score</b> поменям пропущенные значения на заглушку и не забудем перевести столбец в числовой формат: 

In [9]:
games['user_score'] = games['user_score'].replace(
    'tbd', np.NaN).fillna(-1).astype('float')

### столбец *rating*
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [10]:
games['rating'].unique()

array(['E', nan, 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP'], dtype=object)

<font size=2>
Ситуация скорее всего аналогичная столбцу <b>critic_score</b>. Поиск на сайте ESRB ясности не вносит, игры ищутся только по названию, без указания года выхода, процесс достоверного установления значения рейтинга, даже если игра и есть в ESRB будет довольно трудоёмким, возрастной рейтниг у игр также не всегда довольно очевиден, поэтому пропущенные значения заменим на  <i>'no rating'</i>

In [11]:
games['rating'] = games['rating'].fillna('no rating')

### столбцы *year_of_release, name, genre*
<font size="2"><i>([к началу проекта](#К.))</i></font>

<font size=2>
Пускай пропуски в данных столбцах незначительны, но их неверное заполнение может исказить дальнейшие результаты, поэтому ограничимся их заменой на пустые строки = <b>" "</b>, для столбцов с названием и жанром игры. В столбце в годом выхода игры поменяем тип данных, пропуски оставим как есть, т.к заменить значения не представляется возможным.

In [12]:
games.fillna({"name": " ", "genre": " "}, inplace=True)

In [13]:
games['year_of_release'] = games['year_of_release'].astype('Int64')

### Расчёт и добавление в данные суммарных продаж
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [14]:
games['total_sales'] = games['na_sales'] + \
    games['eu_sales'] + games['jp_sales'] + games['other_sales']

<font size=2>
<b>Ещё раз посмотрим информация об исходных данных:</b>

In [15]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 12 columns):
name               16715 non-null object
platform           16715 non-null object
year_of_release    16446 non-null Int64
genre              16715 non-null object
na_sales           16715 non-null float64
eu_sales           16715 non-null float64
jp_sales           16715 non-null float64
other_sales        16715 non-null float64
critic_score       16715 non-null float64
user_score         16715 non-null float64
rating             16715 non-null object
total_sales        16715 non-null float64
dtypes: Int64(1), float64(7), object(4)
memory usage: 1.5+ MB


In [16]:
games.sample(15).style.set_properties(
    **{'text-align': 'center'}).highlight_null(null_color='lightcoral')

<font size=2>
Данные подготовили, избавились от пропусков, поменяли тип данных и добавили необходимые в дальнейшем значения.

In [17]:
games_ps3 = games[(games['platform'] == 'PS3') &
                  (games['critic_score'] > 0)].copy()

# Исследовательский анализ данных
<font size="2"><i>([к началу проекта](#К.))</i></font>

## Выход игр по годам
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [25]:
ggames = games.groupby("year_of_release")["total_sales"].sum().reset_index()
ggames


year_of_release  total_sales
0              1980        11.38
1              1981        35.68
2              1982        28.88
3              1983        16.80
4              1984        50.35
5              1985        53.95
6              1986        37.08
7              1987        21.70
8              1988        47.21
9              1989        73.45
10             1990        49.37
11             1991        32.23
12             1992        76.16
13             1993        48.41
14             1994        79.23
15             1995        88.12
16             1996       199.15
17             1997       201.07
18             1998       256.31
19             1999       251.11
20             2000       201.66
21             2001       331.46
22             2002       394.97
23             2003       357.54
24             2004       418.68
25             2005       457.82
26             2006       517.71
27             2007       604.75
28             2008       671.50
29             2009       658.82
30             2010       590.13
31             2011       507.58
32             2012       355.84
33             2013       361.24
34             2014       331.53
35             2015       267.98
36             2016       129.94

In [29]:
#sns.relplot(x="year_of_release", y="total_sales", kind="line", data=ggames, figsize=(20,10))
sns.histplot(data=ggames)



AttributeError: module 'seaborn' has no attribute 'histplot'

In [ ]:
# посчитаем число вышедших игр и проданных копий по годам
years_games = games.groupby('year_of_release')[['total_sales']].sum().join(
    games.groupby('year_of_release')[['name']].count())
# число проданных копий
trace0 = go.Scatter(x=years_games.index,
                    y=years_games.total_sales, name='Продажи, млн.копий',
                    mode='lines+markers', line_color='indianred')
# число вышедших игр
trace1 = go.Scatter(x=years_games.index,
                    y=years_games.name, name='Кол-во вышедших игр',
                    mode='lines+markers', line_color='mediumseagreen')
# определяем массив данных и задаем title графика в layout
data = [trace0, trace1]
layout = {}
# cоздаем объект Figure и визуализируем его
fig = go.Figure(data=data, layout=layout)

fig.update_layout(
    width=950,
    height=500,
    title='Видеоигры: продажи и выход по годам',
    legend_orientation="h")
iplot(fig, show_link=False)

<font size=2>
До 90-х годов XX века в мире практически не существовало рынка видеоигр, с появлением и распространением компьютеров и современных игровых приставок, рынок начал расти, взрывной рост наблюдается в начале нового тысячелетия в 2002 году, с выходм на пик в 2008/09 годах, но за таким резким подъёмом следует такой же резкий спад, связанный с началом нового периода - периода массового распространения смартфонов, " миграции" людей в соцсети и мобильный гейминг.

## Продажи по платформам
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
platform_games = games.groupby('platform')[['total_sales']].sum()

colors = ['gold', 'mediumturquoise', 'darkorange', 'lightgreen']
fig = go.Figure()
fig.add_trace(go.Pie(
    values=platform_games['total_sales'], labels=platform_games.index, hole=.4))
fig.update_layout(
    width=850,
    height=700,
    title="Продажи по платформам",
    title_x=0.5,
    margin=dict(l=0, r=10, t=50, b=10),
    legend_orientation="v",
    annotations=[dict(text='% продаж<br>в зависимости<br>от платформы',
                      x=0.5, y=0.5, font_size=15, showarrow=False)])
fig.show()

<font size=2>
Можно выделить 6 самых популярных платформ, в порядке убывания: <b>PS2, X360, PS3, Wii, DS, PS</b>. Ожидаемо в лидерах платформа <b>PlayStation</b> и её прямые конкуренты от Microsoft и Nintendo - <b>Xbox360</b> и <b>Wii</b> соответственно.<br/>Можно отметить, что довольно популярный в России компьютерный [гейминг](https://gamemag.ru/news/162765/pandemiya-vernula-interes-k-igram) в общемировом рейтинге только на 10 месте.

## Платформы с наибольшими суммарными продажами
<font size="2"><i>([к началу проекта](#К.))</i></font>
<br/>
<font size=2>
    
<font size=2>Cамая первая версия игровой приставки PlayStation [выпущена](https://ru.wikipedia.org/wiki/PlayStation_(серия_игровых_консолей)#PlayStation) в далёком декабре 1994 года, а на рынок США вышла в сентябре 1995 года. В 2000 году весной в Японии, а осенью в США начались продажи самой популярной игровой консоли в истории - <b>PlayStation 2</b>, продажи преемника, консоли <b>PlayStation 3</b> начались осенью 2005 года.</br> Продажи прямого конкурента приставкам Sony от компании Microsoft -  <b>Xbox 360</b> [стартовали](https://ru.wikipedia.org/wiki/Xbox_360#Начало_продаж) в ноябре 2005 года. Продажи <b>Nintendo DS</b> и <b>Nintendo Wii</b> начались соответственно зимой [2004](https://ru.wikipedia.org/wiki/Nintendo_DS) и -[2006](https://ru.wikipedia.org/wiki/Wii#Продажи) годов соответственно.<br/>
Таким образом мы можем выделить 5 самых популярных игрофых платформ на продажи которых приходится 55%, тогда как на остальные невошедшие в Топ-5 игровые платформы будет приходится 45% от общего числа продаж. Данные возьмём в периоде начиная с 2005 года, да 2000 год, с которого началось активное увеличение и продаж, и количество выхода игр уже не кажется таким актуальным, поскольку появилось уже не одно новое поколение игроков, которым скорее всего не будут интересны игры того периода, которые уже явно не отвечают современным веяниям в части геймдизайна и графики.

In [ ]:
"""оставляем данные с 2005 года"""
games_from_2k = games[games['year_of_release'] >= 2005]
"""формируем Топ-5 платформ"""
top_5 = games_from_2k.groupby(
    'platform')['total_sales'].sum().sort_values()[-5:].index.tolist()
# определяем массив данных и задаем title графика в layout
data = []
for platform in top_5:
    data.append(
        go.Bar(x=games_from_2k[games_from_2k.platform == platform]
               .groupby("year_of_release")['total_sales'].sum().index,
               y=games_from_2k[games_from_2k.platform == platform]
               .groupby("year_of_release")['total_sales'].sum(),
               name=platform)
    )
layout = {}
# cоздаем объект Figure и визуализируем его
fig = go.Figure(data=data, layout=layout)
fig.update_layout(
    width=950,
    height=500,
    title='Продажи по годам Топ-5 платформ',
    title_x=0.5,
    margin=dict(l=10, r=10, t=50, b=10),
    legend_orientation="h")
iplot(fig, show_link=False)

<font size=2>
Изучив график можно заметить, что каждая из игровых платформ в отдельности "живёт" примерно 10 лет. Примерно к середине совего жизненного цикла достигая пика продаж, за которым следует спад.<br/>
На примере истории выхода серии приставок <b>PlayStation</b> мы можем заметить, что компания как раз и старается представлять каждое следующее поколение приставок с интервалом в 5-6 лет, что в общем-то и логично, ведь ведущие производители аппаратных компонентов регулярно и ежегодно представляют и выводят на рынок новые поколения "железа". И даже если конечному потребителю - простому пользователю с его точки зрения прирост производительности будет не так заметен, исключение тут составит категория заядлых геймеров, то вот сам факт обладания приставкой десятилетней давности уже чисто психологически будет настраивать на приобретение нового "железа", особенно если уже вышло новое поколение любимой игровой платформы. Так и игроделы не сидят на месте, выпуская и новые игры и продолжая выпускать уже сложившиеся игровые серии.

## Динамика продаж
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
data = []

for platform in games_from_2k['platform'].unique():
    data.append(
        go.Scatter(x=games_from_2k[games_from_2k.platform == platform]
                   .groupby("year_of_release")['total_sales'].sum().index,
                   y=games_from_2k[games_from_2k.platform == platform]
                   .groupby("year_of_release")[
            'total_sales'].sum(),
            name=platform)
    )

layout = {'title': 'Продажи по платформам'}
# cоздаем объект Figure и визуализируем его
fig = go.Figure(data=data, layout=layout)
fig.update_layout(
    width=950,
    height=700,
    title='Динамика продаж по годам',
    title_x=0.5,
    margin=dict(l=10, r=10, t=50, b=10),
    legend_orientation="h")
iplot(fig, show_link=False)

<font size=2>Продажи падают по всем платформам, даже такие новые платформы как <b>Xbox One</b> и <b>PlayStation 4</b>, вышедшые в 2013 году после стремительно старта демонстрируют не менее стремительный спад уже к 2016 году.<br/> При этом эти платформы в силу их относительной новизны мы момем считать потенциально прибыльными, как и <b>PC</b>, поскольку здесь он демонстрирует не самые плохие показатели. Считать потенциально прибыльными платформы от <b>Nintendo WiiU</b> и <b>3DS</b> я бы не стал, поскольку на текущий момент производителем прекращена их поддержка в [2017](https://ru.wikipedia.org/wiki/Wii_U) и [2020](https://ru.wikipedia.org/wiki/Nintendo_3DS) годах соответственно. Как и остальные платформы, добравшиеся до 2016 года, всё же их жизненный цикл подходит или подошёл к концу.

## Диаграммы размаха по глобальным продажам игр в разбивке по платформам
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
data = []

for platform in games_from_2k['platform'].unique():
    data.append(
        go.Box(y=games_from_2k[games_from_2k.platform == platform]['total_sales'],
               name=platform)
    )
layout = {}
# cоздаем объект Figure и визуализируем его
fig = go.Figure(data=data, layout=layout)
fig.update_layout(
    width=950,
    height=500,
    yaxis=dict(range=[0, 2]),
    title='Диаграммы размаха продаж',
    title_x=0.5,
    margin=dict(l=10, r=10, t=50, b=10),
    legend_orientation="h")
iplot(fig, show_link=False)

<font size=2> В первую очередь иы можем заметить большое количество "выбросов" по всем платформам, исключая только DC - Dreamcast от Sega - приставки со сложной судьбой, бившей [рекорды продаж](https://ru.wikipedia.org/wiki/Dreamcast#Запуск), но так и не ставшей [популярной](https://ru.wikipedia.org/wiki/Dreamcast#Спад_и_прекращение_производства) во всём мире. Наибольшие медианные значения продаж приходятся в большинстве своём на актуальные платформы, а значения выбросов скорее всего показывают продажи хитовых, популярных игр.

## Влияние на продажи внутри популярной платформы отзывов пользователей и критиков
<font size="2"><i>([к началу проекта](#К.))</i></font>

### Диаграммы рассеяния
<font size="2"><i>([к началу проекта](#К.))</i></font>

<font size=2> Посмотрим, как могут влиять на продажи оценки. Посмотрим на примере конкурирующих платформ - <font color='navy'><b>PS4</font></b> и <font color='forestgreen'><b>Xbox One</font></b>.

In [ ]:
# games_from_2k
ps_4 = games_from_2k.query('platform == "PS4"')
xbox_one = games_from_2k.query('platform == "XOne"')

In [ ]:
f, ax = plt.subplots(4, 1, sharex=True, figsize=(20, 25))

ax[0].set_title('Оценки критиков, PS4', size=15)
ax[2].set_title('Оценки пользователей, PS4', size=15)
ax[1].set_title('Оценки критиков, Xbox One', size=15)
ax[3].set_title('Оценки пользователей, Xbox One', size=15)

sns.scatterplot(data=ps_4, x="total_sales", y="critic_score",
                size="critic_score", sizes=(1, 200), color='navy', alpha=0.7, ax=ax[0])
sns.scatterplot(data=ps_4, x="total_sales", y="user_score",
                size="user_score",  sizes=(1, 200), color='navy', alpha=0.7,  ax=ax[2])
sns.scatterplot(data=xbox_one, x="total_sales", y="critic_score",
                size="critic_score", sizes=(1, 200), color='forestgreen', alpha=0.7, ax=ax[1])
sns.scatterplot(data=xbox_one, x="total_sales", y="user_score",
                size="user_score", sizes=(1, 200), color='forestgreen', alpha=0.7, ax=ax[3])

plt.xlabel('Продажи, млн.', size=15)
plt.suptitle("Влияние на продажи", size=20);

<font size=2> Чем выше оценка критиков, тем лучше продажи.А вот с пользовательсткой оценкой не всё так однозначно, нередки продажи игр и с не таким уж и высоким рейтингом. У широких масс игра может быть не такой уж и популярной, что явно скажется на среднем рейтинге, но при этом у неё может быть довольно обширная фанбаза. А может быть и обратная ситуация, когда игра выходит в довольно популярной серии, например те же серии игр Need for Speed, Assassin's Сreed, Battlefield и т.д., которые имеют высокие продажи на старте, в т.ч. за счёт предзаказов, но потом получившие негативный опыт пользователи ставят игре низкие оценки. Тут нельзя не упомянуть уже ставшее, так сказать, притчей во языцех, техническое состояние на момент релиза Cyberpunk 2077.<br/>  А вот от платформы оценки скорее всего не зависят, за редким исключением эксклюзивных изданий, производители игр, если есть такая возможность, всё же стараются издавать игры на всех популярных платформах.

### Матрицы корреляции
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
ps_4 = ps_4[['total_sales', 'critic_score', 'user_score']]
xbox_one = xbox_one[['total_sales', 'critic_score', 'user_score']]

f, ax = plt.subplots(1, 2, sharex=True, figsize=(20, 6))
ax[0].set_title('PS4', size=15)
ax[1].set_title('Xbox One', size=15)

sns.heatmap(ps_4.corr(), annot=True, cmap="Blues", fmt='.2f', ax=ax[0])
sns.heatmap(xbox_one.corr(), annot=True, cmap="Greens", fmt='.2f', ax=ax[1])

plt.xlabel('')
plt.suptitle("Матрицы корреляции", size=20)

<font size=2> Оценки критиков на продажи оказывают вдвое большее влияние, чем оценки пользователей. Стоит также отметить высокую корреляцию этих оценок между собой, что в общем-то логично. Существенное различие этих оценок скорее всего будет говорить о том, что критик знаком с игрой лишь поверхностно, в отличие от пользователя, либо вообще не знаком и его оценка была пролоббирована издателем игры, что на самом деле встречается, причем не только в игровой индустрии.<br/>
    В Топ-5 наших платформ вошли приставки PlayStation 2 и 3 от Sony, Wii и DS от Nintendo, и Microsoft Xbox 360, посмотрим, как выглядит корреляция по ним:

In [ ]:
ps_2 = games_from_2k.query('platform == "PS2"')
ps_2 = ps_2[['total_sales', 'critic_score', 'user_score']]
ps_3 = games_from_2k.query('platform == "PS3"')
ps_3 = ps_3[['total_sales', 'critic_score', 'user_score']]
x360 = games_from_2k.query('platform == "X360"')
x360 = x360[['total_sales', 'critic_score', 'user_score']]
wii = games_from_2k.query('platform == "Wii"')
wii = wii[['total_sales', 'critic_score', 'user_score']]
ds = games_from_2k.query('platform == "DS"')
ds = ds[['total_sales', 'critic_score', 'user_score']]

In [ ]:
f, ax = plt.subplots(5, 1, sharex=True, figsize=(8, 12))
ax[0].set_title('PlayStation 2', size=15)
ax[1].set_title('PlayStation 3', size=15)
ax[2].set_title('Xbox 360', size=15)
ax[3].set_title('Nintendo Wii', size=15)
ax[4].set_title('Nintendo DS', size=15)

sns.heatmap(ps_2.corr(), annot=True, cmap="twilight_r", fmt='.2f', ax=ax[0])
sns.heatmap(ps_3.corr(), annot=True, cmap="twilight_r", fmt='.2f', ax=ax[1])
sns.heatmap(x360.corr(), annot=True, cmap="twilight_r", fmt='.2f', ax=ax[2])
sns.heatmap(wii.corr(), annot=True, cmap="twilight_r", fmt='.2f', ax=ax[3])
sns.heatmap(ds.corr(), annot=True, cmap="twilight_r", fmt='.2f', ax=ax[4])

plt.xlabel('')
plt.suptitle("Матрицы корреляции для популярных платформ", size=20)

<font size=2> Для платформ PS2, PS3 и Xbox 360 данные по корреляции между продажами и оценками даже выше, чем у рассмотренных раньше PS4 и Xbox One, всё же исходные данные по ним скорее всего представлено больше и полнее, в силу их более раннего появления на рынке. Пользователи скорее всего не успели купить PS4 и Xbox One в таком количестве как приставки из Топ-5, поэтому по ним скорее всего и нет такого большого количества разнообразных оценок.<br/>
    Интересна ситуация с приставками Nintendo, с самой низкой корреляцией оценок к продажам. DS хоть и была самой популярной и [продаваемой](https://club.dns-shop.ru/digest/25255-nintendo-ds-samaya-prodavaemaya-konsol-v-ssha-za-poslednie-25-le/) консолью на рынке США, даже обогнав PlayStation 2, специфичиский форм-фактор этой мобильной консоли накладывает ограничения на ассортимент игр и соответствено влияет на кол-во оценок и пользователей, и критиков. Это единственная из рассмотренных приставок, у которой оценки пользователей оказывают большее влияние на продажи, чем оценки критиков, пукай разница в этих значениях не так уж и велика.<br/>
    То же самое мы можем сказать и о Wii - приставке, уникальный контроллер которой [позволял](https://ru.wikipedia.org/wiki/Wii_Remote) игрокам управлять персонажами игры и предметами на экране движением руки, соответственно и большая часть игр для этой приставки, если, возможно не все, были заточены под уникальное взаимодействие платформы с игроками. Да, игры для этой приставки в своём роде уникальны, но таким же разнообразием ассортимента как на других платформах похвастать не могут.


## Распределение игр по жанрам
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
genre_games_sales = games.groupby('genre')[['total_sales']].sum()
genre_games_name = games.groupby('genre')[['name']].count()

fig = make_subplots(rows=1, cols=2, specs=[
                    [{'type': 'domain'}, {'type': 'domain'}]])
fig.add_trace(go.Pie(
    values=genre_games_sales['total_sales'], labels=genre_games_sales.index), 1, 1)
fig.add_trace(go.Pie(
    values=genre_games_name['name'], labels=genre_games_name.index), 1, 2)
fig.update_traces(hole=.6, hoverinfo="label+percent+name")

fig.update_layout(
    width=850,
    height=500,
    title_text="Распределение игр по жанрам",
    title_x=0.5,
    margin=dict(l=0, r=10, t=50, b=10),
    legend_orientation="h",
    annotations=[dict(text='%<br>продаж<br>по жанрам', x=.163, y=.51,
                      font_size=15, showarrow=False),
                 dict(text='% игр<br>по жанрам', x=0.84, y=0.5,
                      font_size=15, showarrow=False)])
fig.show()

<font size=2>Больше половины продаж дают 4 жанра: Action, Sports, Shooter и Role-Playing, причём жанры Action и Sports также лидеры и по количеству изданных игр. Жанр Shooter, тоже вполне ожидаемо в Топе на 3 месте по продажам, но вот по количеству изданных игр это жанр только на 5 месте, можно сказать, что "КПД" этого жанра лучше у двух, расположенных выще - меньшее кол-во игр при большем количестве продаж. Ролевые игры показывают 4 результат как по продажам, так и по количеству. 5 место по продажам за Platform, хотя по количеству представленных игр этот жанр только на 8 месте. <br/>
   

# Портрет пользователя каждого региона
<font size="2"><i>([к началу проекта](#К.))</i></font>

## Самые популярные платформы  (топ-5)
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
# Топ-5 по регионам
def top_region(groupby, region_sales, ax):
    pivot = games.groupby(groupby).agg({region_sales: 'sum'}).sort_values(
        by=region_sales, ascending=False)[:5]
    title_dict = {'na_sales': 'Северная Америка',
                  'eu_sales': 'Европа', 'jp_sales': 'Япония'}
    color_dict = {'na_sales': 'indianred',
                  'eu_sales': 'steelblue', 'jp_sales': 'lightpink'}
    plot = pivot.plot(kind='bar', ec='k', title=title_dict[region_sales], ax=axes[ax],
                      fontsize=18, color=color_dict[region_sales], rot=0)
    plot.legend(prop={'size': 17})
    plot.set_xlabel('')
    plot.title.set_size(20)

In [ ]:
fig, axes = plt.subplots(3, 1, figsize=(17, 15))
plt.suptitle("Популярные платформы", size=20)

top_region('platform', 'na_sales', 0)
top_region('platform', 'eu_sales', 1)
top_region('platform', 'jp_sales', 2)

<font size=2>Среди самых популярных платформ в США большей популярностью пользовались Xbox 360 и PlayStation 2, можно сказать, что они делят между собой 1 и 2 места, Wii на 3 месте по полулярности, а PlayStation 3 и Nintendo DS замыкаю Топ-5, на 4 и 5 местах соответственно.<br/> Ситуация в Европе схожая, только PlayStation 2 здесь на первом месте, PlayStation 3 на втором, а тройку лидеров замыкает Xbox 360, Wii совсем немного отстаёт от неё, а замыкает пятёрку по количеству продаж самая первая версия PlayStation.<br/> Совсем другая ситуация в Японии, здесь не нашлось места компании Microsoft, весь Топ-5 состоит из приставок "отечественного производителя". В лидерах Nintendo DS, 2-3 место делят между собой два первых поколения приставок PlayStation. На четвёртом месте раритет Super Nintendo, что было бы удивительно для остального мира, но не для Японии. И замыкает пятёрку Nintendo 3DS.
   

## Самые популярные жанры (топ-5)
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
fig, axes = plt.subplots(3, 1, figsize=(17, 15))
plt.suptitle("Популярные жанры", size=20)
top_region('genre', 'na_sales', 0)
top_region('genre', 'eu_sales', 1)
top_region('genre', 'jp_sales', 2)

<font size=2>Рассматривая жанры, мы можем увидеть практически полное совпадение вкусов пользователей в Северной Америке и Европе, исключение составляет только 4 место: в Северной Америке это платформеры, а в Европе гоночные игры.<br/> В Японии же опять совершенно другая ситуация - с двукратным отрывом на первом месте расположились ролевые игры, а представители жанра экшн, спорт, платформеры и игры смешанного жанра, с учетом их отставания, можно сказать, что между собой борятся за второе место.

## Влияет ли рейтинг ESRB на продажи в отдельном регионе
<font size="2"><i>([к началу проекта](#К.))</i></font>
<br/>

|Обозначение|рейтинга|Расшифровка|
|:---|:---	|---:	|
|<b>RP|Rating Pending|<i>Рейтинг ожидается|
|<b>E|Everyone|<i>Для всех|
|<b>E10+|Everyone 10 and older|<i>Для всех от 10 лет и старше|
|<b>T|Teen|<i>Подросткам|
|<b>M|Mature|<i>Для взрослых|
|<b>AO|dults Only 18+|<i>Только для взрослых|
|<b>EC|Early childhood|<i>Для детей младшего возраста|
|<b>K-A|Kids to Adults|<i>Для всех(устар.)|
|<b>no rating|  	|<i>рейтинг не указан|

In [ ]:
fig, axes = plt.subplots(3, 1, figsize=(20, 15))
plt.suptitle("Продажи игр в зависимости от рейтинга ESRB", size=20)
top_region('rating', 'na_sales', 0)
top_region('rating', 'eu_sales', 1)
top_region('rating', 'jp_sales', 2)

<font size=2>Топ-5 в Северной Америке и в Европе опять схожи. Первое место за играми <i><b>"Для всех"</i></b>, второе у игр с неуказанным рейтингом, 3 и 4 место в Северной Америке у игр для <i><b>"Для подростков"</i></b> и игр <i><b>"Для взрослых"</i></b>, в Европе продажи игр с таким рейтингом не так близки как и меняются местами, а на пятом месте и там, и там игры <i><b>"Для всех от 10 лет"</i></b>
    <br/> Япония опять не стала изменять себе, опять на этом рынке совершенно другая ситуация. С большим отрывом лидируют продажи игр без рейтинга, но предполагая специфику рынка можно предположить, что в нашем случае в этот список могут попадать как и специфичиские местные игры, которые в лучшем случае получили бы рейтинг <i><b>"Adults only 18+"</i></b> или <i><b>"Mature"</i></b>, так и просто игры, до которых не добрался ESRB - всё же это организация, нацеленная в первую очередь на определение и принятие рейтингов для игр, представленных на рынках США и Канады. А вот по остальным рейтингам ситуация не сильно отличается от остального мира, в порядке убывания продаж в Японии идут жанры  <i><b>"Для всех", "Для подростков", "Для взрослых"</i></b> и <i><b>"Для всех от 10 лет"</i></b>.

# Проверка гипотез
<font size="2"><i>([к началу проекта](#К.))</i></font>

## Средние пользовательские рейтинги платформ Xbox One и PC одинаковые
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
"""Создадит датасеты только с нужными нам для проверки гипотезы данными"""
x_one_games = (games.query('platform=="XOne"').pivot_table(
    index='name', values=('user_score'))).reset_index()
pc_games = (games.query('platform=="PC"').pivot_table(
    index='name', values=('user_score'))).reset_index()

"""убираем заглушки ("-1"), чтобы не искажать результаты и построить нормальные графики"""
x_one_games = x_one_games[x_one_games['user_score'] > 0]
pc_games = pc_games[pc_games['user_score'] > 0]

### Тест Левена
<font size="2"><i>([к началу проекта](#К.))</i></font>
<br/> <font size=2> Перед проверкой гипотезы посмотрим на дисперсии пользовательских оценок и сравним наши выборки с помощью теста Левена:

In [ ]:
print('Дисперсия пользовательских оценок на Xbox One составляет: {:,.2f}'
      .format(x_one_games['user_score'].var()))
print('Дисперсия пользовательских оценок на PC составляет: {:,.2f}'
      .format(pc_games['user_score'].var()))
print('Дисперсии отличаются на: {:,.2%}'
      .format((pc_games['user_score'].var()/x_one_games['user_score'].var() - 1)))

In [ ]:
alpha = .05
p_value = st.levene(x_one_games['user_score'], pc_games['user_score'])
p_value

<font size=2> Значение p-критерия достаточно велико, нулевую гипотезу теста Левена - одинаковость дисперсии между нашими выборками, можно считать подтверждённой.

### Проверка гипотезы
<font size="2"><i>([к началу проекта](#К.))</i></font>
<br/> <font size=2> <b>H0 (нулевая гипотеза)</b> - предположение о том, что средние оценки пользователей равны.<br/>
<b>H1 (альтернативная гипотеза)</b> - средние оценки отличаются.<br/>
<b>α=5%</b>
<b>equal_var=True</b>, т.к. дисперсии выборок одинаковы.

In [ ]:
alpha = .05
results = st.ttest_ind(
    x_one_games['user_score'], pc_games['user_score'], equal_var=True)
print('P-value:', results.pvalue)

if (results.pvalue < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось опровергнуть нулевую гипотезу")

<font size=2> Значение p-критерия стремиться к нулю, можно отклонить нулевую гипотезу и утверждать, что средние пользовательские оценки платформ <b>Xbox One</b> и <b>PC</b> отличаются между собой. Посмотрим на оценки более подробно:

In [ ]:
fig = go.Figure()
fig.add_trace(go.Box(x=x_one_games['user_score'],
                     fillcolor='palegreen', line=dict(color='olivedrab'),
                     notched=True, boxmean=True, name='Xbox One', boxpoints='all'))
fig.add_trace(go.Box(x=pc_games['user_score'],
                     fillcolor='sandybrown', line=dict(color='chocolate'),
                     notched=True, boxmean=True, name='PC', boxpoints='all'))
# fig.update_traces(quartilemethod="inclusive")
fig.update_layout(
    width=950,
    height=700,
    title="Диаграммы размаха пользовательских оценок платформ Xbox One и PC",
    title_x=0.5,
    yaxis_title="Платформа",
    legend=dict(x=.5, xanchor="center", orientation="h", font_size=15),
    barmode='overlay',
    margin=dict(l=10, r=20, t=50, b=10))

fig.show()

<font size=2> Пользователи охотнее ставят оценки играм на <b>PC</b> и эти оценки тяготеют к более высоким значениям - средняя пользовательская оценка почти <b>7,1 балла</b>, и это хорошая, положительной оценка игры.<br/>
    Игры на <b>Xbox One</b> не могут похвастать таким же разнообразием и количеством оценок, средняя оценка даже не дотягивает до 7 баллов, и составляет <b>6,5 баллов</b>, безусловно это ещё не такая низкая и критичная оценка, но всё же это уже не твёрдые <i>7 из 10</i>. Действительно, средняя оценка игр на платформах отличается.<br/>
    И если верить пользователям, то в целом на <b>PC</b> качество игр в выше, чем на <b>Xbox One</b>.<br/>
    <i>Большее количество оценок можно объяснить не только, возможно, большим разнообразием игр на PC, но и банально удобством выставления оценки: приставки в первую очередь заточены под игры и пользователям необходимо провести большее число манипуляций, чтобы поставить оценку игре и/или написать отзыв, в отличие от PC, где если игра не понравилась, можно довольно быстро зайти в интернет и поставить оценку. </i>

## Средние пользовательские рейтинги жанров Action и Sports разные
<font size="2"><i>([к началу проекта](#К.))</i></font>

In [ ]:
"""Создадим датасеты только с нужными нам для проверки гипотезы данными"""
action_games = (games.query('genre=="Action"').pivot_table(
    index='name', values=('user_score'))).reset_index()
sport_games = (games.query('genre=="Sports"').pivot_table(
    index='name', values=('user_score'))).reset_index()

"""убираем заглушки ("-1"), чтобы не искажать результаты и построить нормальные графики"""
action_games = action_games[action_games['user_score'] > 0]
sport_games = sport_games[sport_games['user_score'] > 0]

### Тест Левена
<font size="2"><i>([к началу проекта](#К.))</i></font>
<br/> <font size=2> Перед проверкой гипотезы посмотрим на дисперсии пользовательских оценок и сравним наши выборки с помощью теста Левена:

In [ ]:
print('Дисперсия пользовательских оценок жанра "Action" составляет: {:,.2f}'
      .format(action_games['user_score'].var()))
print('Дисперсия пользовательских оценок жанра "Sports" составляет: {:,.2f}'
      .format(sport_games['user_score'].var()))
print('Дисперсии отличаются на: {:,.2%}'
      .format((sport_games['user_score'].var()/action_games['user_score'].var() - 1)))

In [ ]:
alpha = .05
p_value = st.levene(action_games['user_score'], sport_games['user_score'])
p_value

<font size=2> Значение p-критерия достаточно мало, чтобы мы могли утверждать одинаковость дисперсии между нашими выборками, будем считать их отличающимися.

### Проверка гипотезы
<font size="2"><i>([к началу проекта](#К.))</i></font>

<br/> <font size=2> <b>H0 (нулевая гипотеза)</b> - предположение о том, что средние оценки пользователей разные.<br/>
<b>H1 (альтернативная гипотеза)</b> - средние оценки одинаковы.<br/>
<b>α=5%</b>
<b>equal_var=False</b>, т.к. дисперсии выборок одинаковы.

In [ ]:
alpha = .05
results = st.ttest_ind(
    action_games['user_score'], sport_games['user_score'], equal_var=False)
print('P-value:', results.pvalue)

if (results.pvalue < alpha):
    print("Отвергаем нулевую гипотезу")
else:
    print("Не получилось опровергнуть нулевую гипотезу")

<font size=2> Значение p-критерия не настолько мало, как в случае с проверкой гипотезы по платформам, но всё же оно мало достаточно, чтобы отклонить нулевую гипотезу и утверждать, что средние пользовательские оценки жанров <b>Action</b> и <b>Sports</b> равны. Посмотрим на оценки более подробно:

In [ ]:
fig = go.Figure()
fig.add_trace(go.Box(x=action_games['user_score'],
                     fillcolor='cornflowerblue', line=dict(color='royalblue'),
                     notched=True, boxmean=True, name='Action', boxpoints='all'))
fig.add_trace(go.Box(x=sport_games['user_score'],
                     fillcolor='tan', line=dict(color='olive'),
                     notched=True, boxmean=True, name='Sports', boxpoints='all'))
# fig.update_traces(quartilemethod="inclusive")
fig.update_layout(
    width=950,
    height=700,
    title="Диаграммы размаха пользовательских оценок жанров Action и Sports",
    title_x=0.5,
    yaxis_title="Жанр",
    legend=dict(x=.5, xanchor="center", orientation="h", font_size=15),
    barmode='overlay',
    margin=dict(l=10, r=20, t=50, b=10))

fig.show()

<font size=2> Да, для игр жанра <b>Sports</b> средняя оценка немногим больше 6 баллов, а вот у жанра <b>Action</b> немного не дотягивает до 6,5 баллов. В целом можно сказать, что да, оценки примерно равны. Также как и при сравнении оценок по платформам, в распределении оценок по жанрам мы можем видеть, что у жанра <b>Action</b> гораздо больше пользовательских оценок, по сравнению с жанром <b>Sports</b>. Это хорошо было заметно ещё на этапе исследовательского анализа данных - игр в жанре <b>Action</b> больше, больше и продаж в этом жанре и сами игры этого жанра популярнее <b>Sports</b> у пользователей во всех регионах.   
   

# Общий вывод
<font size="2"><i>([к началу проекта](#К.))</i></font>

<font size=2> 
<b><i>Рынок игровой индустрии меняется.</b></i>

Проведённый анализ рынка показал, что индустрия стационарных игровых платформ сейчас переживает не лучшие времена. Золотым период можно назвать начало XXI века, первые 8 лет нового тысячелетия отметились стремительным ростом индустрии, рынок рос фантастическими темпами, причём росло как и количество выпускаемых игр, так и их продажи. Но всё закончилось также быстро, как и началось, если в 2009 году продажи просели совсем немного по сравнению с 2008, то в 2010 году наметился ощутимый спад - меньше и новых игр, и продаж. Винить в этом в первую очередь можно мобильный гейминг, расцвет смартфонов, как раз и пришедшийся на начало второго десятилетия XXI века безусловно изменил весь рынок, ведь теперь возможность поиграть всегда под рукой.<br/>
    В качестве ещё одной причины стагнации рынка можно назвать погоню за технологиями - лидеры индустрии всё чаще и чаще просто гонятся за технологиями. Растут бюджеты, игры штампуются как на конвейере, сериями из года в год, при этом производители забывают о такой важной составляющей геймдизайна как атмосфера в игре, а порой ещё даже забывая об оптимизации, сейчас уже совсем нередки "патчи первого дня". Глубокие и атмосферные игры сейчас удел небольших и независимых студий. Но винить в этом целиком и полностью только производителей игр было бы слишком опрометчиво, как никак они явно тоже следят за тенденциями и их мнение, что игроки ценят только красивую картинку не появилось на пустом месте. Мир меняется, причём не только игровой, меняется сама модель потребления контента и игровой рынок скорее всего продолжит падение, поскольку тендеций к изменению модели потребления не наблюдается. <i>Хотя возможно, что с 2020 года мог наметиться рост продаж немобильнызх игр, поскольку небезызвестная пандемия многих сделала вынужденными домоседами.</i>

Рассматривая же продажи по платформам то актуальными и прибыльными можно считать свежие версии приставок <b>PlayStation</b> от Sony и <b>Xbox One</b> от Microsoft, а также <b>Nintendo Switch</b>, пускай в исходных данных игр по данной консоли нет, но по состоянию на [сентябрь 2021 года](https://www.nintendo.co.jp/ir/en/finance/hard_soft/index.html) компания продала почти 100 млн. приставок, а количество копий проданных игр стремится к 700 млн. И это при том, что на этой консоли представлены в болшьшинстве своём не топовые игры с больших консолей, а только свои эксклюзивы.
    
Для актуальных и популярных игровых платформ влияние на уровень продаж оценок критиков невелико, но всё же больше, чем влияние оценок пользователей. Стоит отметить, что две эти оценки между собой связаны гораздо сильнее, несмотря на то, что критики оценивают игры как правило перед релизом или на самом старте продаж, а пользователи уже получив игровой опыт. Пускай критики не всегда беспристрастны, ведь производители игр заинтересованы в хорошем старте продаж, а значит и хороших оценках от игровых критиков, все участники этой "игры" прекрасно понимают, что если продукт получился "не очень", то высокие оценки критиков его не спасут. Критики, кмк, стараются балансировать между геймерами и издателями и выставлять пускай и высокие, но разумные оценки.
    
В игровых рынках существенных различий мы не наблюдаем, что на рынке Северной Америке, что Европы популярны как типы игровых платформ, так и жанры игр. И рейтинг ESRB тут сюрпризов не преподносит, больше всего продаётся игр для всех возрастных категорий, следом за ним идут игры для тинэйджеров и взрослых. Особняком тут стоит только рынок Японии, где с огромным отрывом лидируют ролевые игры и зная специфику региона можно догадаться, что это игры которые и были выпущены только на домашнем рынке, поскольку у большей части этих игр в представленных данных нет оценки от ESRB.
    
Сравнив же рейтинги по платформам и жанрам мы можем сказать, что игры для <b>PC</b> получают в среднем более высокие оценки от пользователей, чем игры для <b>Xbox One</b>. А вот жанру <b>Action</b> пользователи ставят примерно такие же оценки как и играм в жанре <b>Sports</b>.